src/relax/op/tensor/manipulate.cc

```diff
@@ -46,72 +46,76 @@ StructInfo InferStructInfoBroadcastTo(const Call& call, const BlockBuilder& ctx)
-          << old_ndim - i - 1 << " is " << old_len
-          << ", where the broadcastability cannot be determined at compile-time. If the old shape "
-             "at this dim is symbolic and will be 1 at runtime, to ensure safety, please use "
-             "MatchCast to explicitly make the shape at this dimension be static 1.");
-    }
-  }
+  // if (data_sinfo == nullptr) {
+  //  ctx->ReportFatal(
+  //      Diagnostic::Error(call)
+  //      << "broadcast_to requires the input data to be Tensor. However, the given one is "
+  //      << call->args[0]->struct_info_->GetTypeKey());
+  //}
+  // if (tgt_shape_sinfo == nullptr) {
+  //  ctx->ReportFatal(
+  //      Diagnostic::Error(call)
+  //      << "broadcast_to requires the input new shape to be Shape. However, the given one is "
+  //      << call->args[1]->struct_info_->GetTypeKey());
+  //}
+
+  // if (!data_sinfo->IsUnknownNdim() && !tgt_shape_sinfo->IsUnknownNdim() &&
+  //    tgt_shape_sinfo->ndim < data_sinfo->ndim) {
+  //  ctx->ReportFatal(Diagnostic::Error(call)
+  //                   << "broadcast_to expects the input shape to have the number of ndim at least
+  //                   "
+  //                      "as the input tensor's. However, the given tensor has ndim "
+  //                   << data_sinfo->ndim << " while the target shape has ndim "
+  //                   << tgt_shape_sinfo->ndim);
+  //}
+
+  //// Trust the input target shape when there is no possibility to do any compile-time check.
+  // if (!data_sinfo->shape.defined()) {
+  //  return TensorStructInfo(/*shape=*/call->args[1], data_sinfo->dtype);
+  //}
+  // ShapeStructInfo shape_sinfo =
+  // Downcast<ShapeStructInfo>(data_sinfo->shape.value()->struct_info_); if
+  // (!shape_sinfo->values.defined() || !tgt_shape_sinfo->values.defined()) {
+  //  return TensorStructInfo(/*shape=*/call->args[1], data_sinfo->dtype);
+  //}
+
+  // arith::Analyzer* analyzer = ctx->GetAnalyzer();
+  // Array<PrimExpr> old_shape_value = shape_sinfo->values.value();
+  // Array<PrimExpr> tgt_shape_value = tgt_shape_sinfo->values.value();
+  // int old_ndim = old_shape_value.size();
+  // int tgt_ndim = tgt_shape_value.size();
+  // for (int i = 0; i < old_ndim; ++i) {
+  //  PrimExpr old_len = old_shape_value[old_ndim - i - 1];
+  //  PrimExpr tgt_len = tgt_shape_value[tgt_ndim - i - 1];
+  //  const auto* old_len_int = old_len.as<IntImmNode>();
+  //  if (old_len_int != nullptr && old_len_int->value == 1) {
+  //    continue;
+  //  } else if (!analyzer->CanProveEqual(old_len, tgt_len)) {
+  //    // We would like to ensure safety, and therefore placed a stronger requirement for user to
+  //    // use MatchCast.
+  //    // Todo(relax-team): At this moment, enforcing MatchCast is fine. But we may need to revisit
+  //    // this requirement to reduce the workload of importers and better support dynamic shapes.
+  //    ctx->ReportFatal(
```

Comment:
likely you do not need to update this part with match cast and shape being explicitly passed in amd bring most of the canges back

Reply:
this is a quick change that worth bring to tlc, cc @YuchenJin

Reply:
Yes sorry, that should have been readded. It works fine now.